src/core/exec-invoke.c

```diff
@@ -5505,7 +5510,7 @@ int exec_invoke(
          * running this service might have the correct privilege to change to the working directory. Also, it
          * is absolutely 💣 crucial 💣 we applied all mount namespacing rearrangements before this, so that
          * the cwd cannot be used to pin directories outside of the sandbox. */
-        r = apply_working_directory(context, params, runtime, home);
+        r = apply_working_directory(context, params, runtime, home, accum_env);
```

Comment:
I'd prefer updating the home var after setting up pam, instead of special case this in apply_working_directory

Reply:
i actually wrote it like that first, but then i figured: it's not just PAM that has an effect on the env block, it's various other things too. For example we support Environment= for setting arbitrary variables, and if you set $HOME with that, we should support it for the cwd too. i.e. even if you have a service without PAM, and set $HOME via Environment= in it I am very sure we should honour that, as the concept of least surprise. and this way we do.

Reply:
I mostly mean that this should reflect directly on home var, not leaving the decision for apply_working_direcory

Reply:
i renamed the 'home' variable to 'pwent_home' now, to emphasize it only carries the home dir according to struct passwd, and separate it conceptually from the home configured in $HOME. This means we can now nicely inside of apply_working_directory look at the env var and then at the pw entry fallback, like how we usually do it.

i hope that addresses your concerns enough?